<!DOCTYPE html>
<html lang="ru">

<head>
    <meta charset="UTF-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Обработка ошибок через try catch</title>
    <script>
        "use strict";
        function div(x, y) {
            let result;
            try {
                if (y == 0) throw new Error(`Деление на ноль`);
                result = x / y;
                return result;
            }
            catch (e) {
                console.log(`Ошибка: ` + e.name);
                console.log(e);
            }
            finally {
                console.log(`В любом случае, finally выполняется `)
            }
        }

        let d = div(10, 5);
        console.log(d);
        console.log(div(10, 0));

        window.onerror = function (msg, url, line){
            alert(`Ошибка ${msg} на ${line} строке в файле ${url}`);
        }
        x = 10;
    </script>
</head>

<body>

</body>

</html>